api/domain_handlers.go

```diff
@@ -381,3 +382,70 @@ func (api API) Remove(w http.ResponseWriter, r *http.Request) {
 
 	writeJSONOrBust(w, issues)
 }
+
+// RemoveIneligibleDomains runs eligibility checks on domains present in the
+// database and change the status to PendingAutomatedRemoval if the domain
+// does not follow the requirements for more than 2 crawls
+
+// Example: GET /removeineligibledomains?
+func (api API) RemoveIneligibleDomains(w http.ResponseWriter, r *http.Request) {
+
+	var ineligibleDomains []database.IneligibleDomainState
+	var deleteEligibleDomains []string
+
+	domains, err := api.database.AllDomainStates()
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not retrieve domains. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+	// Store ineligible domains in slice
+	for _, d := range domains {
+		if d.Policy == preloadlist.Bulk18Weeks || d.Policy == preloadlist.Bulk1Year {
+
+			_, issues := api.hstspreload.EligibleDomain(d.Name, d.Policy)
+			state, err := api.database.GetIneligibleDomainStates([]string{d.Name})
+			if len(issues.Errors) > 0 {
+				if len(state) > 0 {
+					state[0].Scans = append(state[0].Scans, database.Scan{
+						ScanTime: time.Now(),
+						Issues:   issues,
+					})
+					ineligibleDomains = append(ineligibleDomains, state[0])
+				} else {
+					ineligibleDomains = append(ineligibleDomains, database.IneligibleDomainState{
+						Name: d.Name,
+						Scans: []database.Scan{
+							{
+								ScanTime: time.Now(),
+								Issues:   issues,
+							},
```

Comment:
After the call to `EligibleDomain`, you can create the `database.Scan` struct with the time instead of needing to repeat this.